pallets/capacity/src/lib.rs

```diff
@@ -1045,26 +1045,30 @@ impl<T: Config> Pallet<T> {
 	pub(crate) fn has_unclaimed_rewards(account: &T::AccountId) -> bool {
 		let current_era = Self::get_current_era().era_index;
 		match Self::get_staking_history_for(account) {
-			Some(provider_boost_history) => {
-				match provider_boost_history.count() {
-					0usize => false,
-					1usize => {
-						// if there is just one era entry and:
-						// it's for the previous era, it means we've already paid out rewards for that era, or they just staked in the last era.
-						// or if it's for the current era, they only just started staking.
-						provider_boost_history
-							.get_entry_for_era(&current_era.saturating_sub(1u32.into()))
-							.is_none() && provider_boost_history
-							.get_entry_for_era(&current_era)
-							.is_none()
-					},
-					_ => true,
-				}
-			},
+			Some(provider_boost_history) =>
+				Self::has_unclaimed_rewards_with_params(current_era, &provider_boost_history),
 			None => false,
 		} // 1r
 	}
 
+	pub(crate) fn has_unclaimed_rewards_with_params(
+		current_era_index: u32,
+		boost_history: &ProviderBoostHistory<T>,
+	) -> bool {
+		match boost_history.count() {
```

Comment:
I creating a variable for naming:
let previous_era =  &current_era_index.saturating_sub(1u32.into())